src/bin/collapse-perf.rs

```diff
@@ -18,33 +24,30 @@ use inferno::collapse::Collapse;
     If you save this output add --header on Linux >= 3.14 to include perf info."
 )]
 struct Opt {
-    /// include PID with process names [1]
-    #[structopt(long = "pid")]
-    include_pid: bool,
-
-    /// include TID and PID with process names [1]
-    #[structopt(long = "tid")]
-    include_tid: bool,
-
-    /// include raw addresses where symbols can't be found
+    // Flags...
+    /// Include raw addresses where symbols can't be found
     #[structopt(long = "addrs")]
-    include_addrs: bool,
+    _addrs: bool,
```

Comment:
I don't understand why these all now start with `_`? Is that a `structopt` thing?

Reply:
So `structopt` orders what the user sees with `--help` in the following way...

* It groups things in the these groups (and in this order): "flags", "options", "args"
* And, more importantly for this comment, within each group, it bases the order of the display on the name of the field in the `Opt` struct, even though what is actually displayed comes from the `long` attribute.

So, at a minimum, I think you want the struct field name to correspond with the `long` attribute.

On the current master branch, the perf flags are displayed like this...

```bash
        --all        all annotations (--kernel --jit)
        --jit        annotate jit functions with a _[j]
        --kernel     annotate kernel functions with a _[k]
    -h, --help       Prints help information
        --addrs      include raw addresses where symbols can't be found
        --pid        include PID with process names [1]
        --tid        include TID and PID with process names [1]
    -q, --quiet      Silence all log output
    -V, --version    Prints version information
    -v, --verbose    Verbose logging mode (-v, -vv, -vvv)
```

which is a bit weird  (like, why is `addrs` all the way in the middle?). It's because the struct fields and the `long` attributes on master are not in sync.

In the current version, the underscores are there because I thought it would be a good idea to have substantive options, like...

```bash
        --addrs      Include raw addresses where symbols can't be found
        --all        All annotations (--kernel --jit)
        --jit        Annotate jit functions with a _[j]
        --kernel     Annotate kernel functions with a _[k]
        --pid        Include PID with process names
        --tid        Include TID and PID with process names
```

come before the "included in basically every CLI application" options like...

```bash
    -h, --help       Prints help information
    -q, --quiet      Silence all log output
    -V, --version    Prints version information
    -v, --verbose    Verbose logging mode (-v, -vv, -vvv)
```

If you don't like that, I would suggest we just do pure alphabetical (which would get rid of the underscores but still be a change from master). What do you think?

Reply:
Ohhhh, I see. The `_` lexicographically sorts before none`_` names. This seems like something `structopt` should fix (is there a tracking issue?), but I'd be okay including this hack for now. Maybe file an issue if one doesn't exist and then link to it from a comment before the first `_` field?

Also, does this change need to be applied to the other binaries as well?

Reply:
It's a feature of `structopt`, not a bug. It decouples the "order in which things show up" to "the names displayed," but it should probably be documented / explained more clearly.

The change probably needs to be applied to other binaries. I'll look and, if so, make a commit for that.